sqlalchemy_utils/functions/database.py

```diff
@@ -459,41 +461,54 @@ def sqlite_file_exists(database):
 
     url = copy(make_url(url))
     database, url.database = url.database, None
-    engine = sa.create_engine(url)
+    engine = None
+    dialect_name = url.get_dialect().name
 
-    if engine.dialect.name == 'postgresql':
+    if dialect_name == 'postgresql':
+        ret = False
         text = "SELECT 1 FROM pg_database WHERE datname='%s'" % database
-        return bool(get_scalar_result(engine, text))
-
-    elif engine.dialect.name == 'mysql':
+        if postgres_db is None:
+            postgres_db = ('postgres', 'template0', 'template1', None)
+        for pdb in postgres_db:
+            url.database = pdb
+            engine = sa.create_engine(url)
+            try:
+                ret = bool(get_scalar_result(engine, text))
+            except (ProgrammingError, OperationalError):
+                pass
+            engine.dispose()
+            if ret:
+                return ret
```

Comment:
I think we should exit the for loop also when `ret` gets set to `False`.
```suggestion
                ret = bool(get_scalar_result(engine, text))
                break
            except (ProgrammingError, OperationalError):
                pass
            finally:
                engine.dispose()
```